Adaptors/S3/src/ObjectStorage.cs

```diff
@@ -274,21 +268,21 @@ internal static class AmazonS3ClientExt
     return fileContent;
   }
 
-  internal static async Task<string?> StringGetValueAsync(this AmazonS3Client    s3Client,
-                                                          string                 bucketName,
-                                                          string                 key,
-                                                          ILogger<ObjectStorage> logger)
+  internal static async Task<string?> StringGetValueAsync(this AmazonS3Client s3Client,
+                                                          string              bucketName,
+                                                          string              key,
+                                                          CancellationToken   cancellationToken = default)
   {
     var response = await s3Client.GetObjectAsync(bucketName,
-                                                 key);
+                                                 key,
+                                                 cancellationToken);
 
     // Get the data from the response stream
     await using var responseStream = response.ResponseStream;
 
     var retrievedData = new byte[responseStream.Length];
-    _ = await responseStream.ReadAsync(retrievedData,
-                                       0,
-                                       retrievedData.Length);
+    _ = await responseStream.ReadAsync(new Memory<byte>(retrievedData),
```

Comment:
I think you can rely on implicit conversion here:
```suggestion
    _ = await responseStream.ReadAsync(retrievedData,
```